SwiftGit2/Objects.swift

```diff
@@ -37,13 +38,35 @@ public struct Signature {
 	/// The time zone that `time` should be interpreted relative to.
 	public let timeZone: TimeZone
 
+	/// Create an instance with custom name, email, dates, etc.
+	public init(name: String, email: String, time: Date = Date(), timeZone: TimeZone = TimeZone.autoupdatingCurrent) {
+		self.name = name
+		self.email = email
+		self.time = time
+		self.timeZone = timeZone
+	}
+
 	/// Create an instance with a libgit2 `git_signature`.
 	public init(_ signature: git_signature) {
 		name = String(validatingUTF8: signature.name)!
 		email = String(validatingUTF8: signature.email)!
 		time = Date(timeIntervalSince1970: TimeInterval(signature.when.time))
 		timeZone = TimeZone(secondsFromGMT: 60 * Int(signature.when.offset))!
 	}
+
+	/// Return an unsafe pointer to the `git_signature` struct.
+	/// Caller is responsible for freeing it with `git_signature_free`.
+	var unsafeSignature: Result<UnsafeMutablePointer<git_signature>, NSError> {
```

Comment:
It think we should probably make this a `makeX` function. That'd be more in line with the Swift API guidelines. `makeUnsafeSignature()`. Or maybe even `makeGitSignature()`.

Reply:
`makeUnsafeSignature()` sounds more consistent with our current `unsafe`-prefix stuff